### ISSUE TITLE
Consume new shot data over network

### ISSUE THREAD

**Issue by andrewnagata** [christianrehn/CombineTest#2]
Hello there! Amazing work here, thanks for sharing!

I am wondering if there is a way to capture shots from a GC3/4 over a network connection. I would like to create an iPad app that listens for each new shot as they are hit.

It seems you are looking for a .csv that gets generated by the Foresight app, and I'm wondering if there are different ways to "listen" for shots.

Tough to find any info on this, and I'm trying to figure this out before spending the $$$ on a unit. Any thoughts would be appreciated. Thanks again!

**Comment by christianrehn** [comment]
Hi, I have written a Java program that reads all the traffic sent over a LAN connection when the GC4 is connected via Ethernet. I think USB or WLAN is very similar.
My problem was that the calculated data (like carry or total) is missing because it is calculated from the measured data by a ball flight model. In the messages sent over the LAN connection include only the measured values.
So I use the data calculated by FSX2020 from the CSV or session data because I have no ball flight model implementation and there is no good free implementation available (at least I could not find one).
If you are only interested in the "new shot event" I could share my Java code with you.

**Comment by andrewnagata** [comment]
Oh, that is interesting, and good to know. So, the number for total carry on the LCD must be derived from their ball flight model on the device, but not sent along with all the measured data. The client on the other end must have its own model, or similar, to derive those numbers. Probably by design to prevent people from doing exactly what I want to do. Ha.

I used to do a bit of Java, and would love to see how you consume those events.

I was planning on using Unity3D/C# to build out some games/experiences, but I think the method to consume events would be similar. Anything you can share would be really helpful, and appreciated.

**Comment by christianrehn** [comment]
Yes, you only get the measured values, everything else is calculated by the firmware or FSX2020, FSX Play, ...
When you compare the values for example for carry distance you will find differences, Foresight seems to have different implementations for the calculation of those values.

If you try to implement something under iOS you cannot access the data calculated by the ball flight models. That is the reason why I implemented my program under Windows. 

You can find my Java Implementation to read the data that is sent by GC4 over Ethernet under https://github.com/christianrehn/gc4connect

**Comment by andrewnagata** [comment]
Ok. thanks! this is really helpful.

Will check out your implementation.

I'm also going to take a crack at the driver score calculation for the Combine using all your data. Happy to share what I come up with

**Comment by christianrehn** [comment]
Would be great if you have a solution for the driver score. 
I think you have already found all Trackman data that I have collected under https://github.com/christianrehn/CombineTest/blob/master/TrackmanData/TestCenterAndCombineReportResults.xlsx
All other scores you need the carry values but I guess for the driver score the total values. 
The data I collect here in Germany is normally in meters not yards but this should not make a difference if you don't mix the units.

**Comment by christianrehn** [comment]
And please let me know when you find the source code of a good ball flight model implementation. 
I don't have the time to reverse engineer the computation of carry, total, side, ... from one of the FSX applications or the firmware.
